includes/Services/FlowService.php

```diff
@@ -0,0 +1,199 @@
+				if(strcmp(gettype($value), gettype($params[$key])) != 0) {
+					$mismatch_data_type = $key. ' => '. gettype($params[$key]) . '. Expected: ' . gettype($value);
+					break;
+				}
+
+				// Accepts non-Array Values entered by the user
+				elseif(!is_array($value))
+					$flow_data[$key] = $params[$key];
+
+				else {
+					// To handle Indexed Arrays gracefully
+					if (count(array_filter(array_keys($params[$key]), 'is_string')) === 0 ) {
+						// If the Database value is empty or an indexed Array, to avoid Associative arrays to be overwritten (Eg: data)
+						((sizeof($value) === 0) || (count(array_filter(array_keys($value), 'is_string')) === 0))?
+							$flow_data[$key] = $params[$key]
+							: $flow_data[$key] = $value;
+						continue;
+					}
+
+					// To handle Associative Arrays gracefully
+					$flow_data[$key] = self::update_post_call_data_recursive($value, $params[$key]);
+				}	
+			}
+
+			// if there is an empty value in the DB or any key is not sent by user, the default value (if any) or the existing DB value is retained
+			else
+				$flow_data[$key] = $value;						
+		}
+
+		return (!empty($mismatch_data_type))? $mismatch_data_type : $flow_data;
+	}
+
+    /*
+	 * Read onboarding flow options from database
+	*/
+	public static function read_flow_data_from_wp_option() {
+		return \get_option( Options::get_option_name( 'flow' ), false );
+	}
+
+    /**
+	 * check the current flow type and update default data if flowtype is ecommerce.
+	 *
+	 * @param default flow data.
+	 *
+	 * @return array
+	 */
+	private static function update_default_data_for_ecommerce( $data ) {
+		// update default data with ecommerce data
+		$data['data']['topPriority']['priority1'] = 'selling';
+		$data['data']['siteType']['referTo'] = 'business';
+		return $data;
+	}
+
+    /*
+	 * function to search for key in array recursively with case sensitive exact match
+	 */
+	public static function check_key_in_nested_array( $params, $flow_data, $header_key = 'Base Level' ) {
+		static $mismatch_key = [];
+		$exception_list = Flows::get_exception_list();
+		foreach($params as $key => $value){
+			if(!isset($exception_list[$key]) && is_array($flow_data)) {
+
+				// Error if the key added by the user is not present in the database
+				if(!array_key_exists($key, $flow_data))
+					$mismatch_key[]  = $header_key. " => " . $key;
+				
+				// To check sub-Arrays
+				elseif ( is_array( $value ) && !empty($value)) {
+					// For Indexed Arrays
+					if (count(array_filter(array_keys($value), 'is_string')) === 0) {
+						foreach($value as $index_key=>$index_value) {
+							// For Indexed Arrays having Associative Arrays as Values
+							if(is_array($index_value))
+								self::check_key_in_nested_array($value[$index_key], $flow_data[$key][$index_key], $key. " : " . $index_key);
+						}
+						continue;
+					}
+
+					// For Associative Arrays
+					self::check_key_in_nested_array($value, $flow_data[$key], $key);
+				}
+			}
+		}
```

Comment:
```suggestion
foreach($params as $key => $value){
			if(isset($exception_list[$key]) || ! is_array( $flow_data ) ) {
				continue;
			}

			// Error if the key added by the user is not present in the database
			if(!array_key_exists($key, $flow_data)) {
				$mismatch_key[]  = $header_key. " => " . $key;
				continue;
			}
			
			// To check sub-Arrays
			if ( ! is_array( $value ) || empty($value)) {
				continue;
			}

			// For Indexed Arrays
			if (count(array_filter(array_keys($value), 'is_string')) === 0) {
				foreach($value as $index_key=>$index_value) {
					// For Indexed Arrays having Associative Arrays as Values
					if(is_array($index_value))
						self::check_key_in_nested_array($value[$index_key], $flow_data[$key][$index_key], $key. " : " . $index_key);
				}
				continue;
			}

			// For Associative Arrays
			self::check_key_in_nested_array($value, $flow_data[$key], $key);
		}
```
Again, lets make the code itself readable

Reply:
Have modified the function accordingly, the changes above suggested were throwing errors for the values of the `$exception_list` keys, hence made changes as required.